fhir-model/src/main/java/com/ibm/fhir/model/util/ValidationSupport.java

```diff
@@ -298,6 +300,105 @@ public static void checkLanguageCodeableConcept(CodeableConcept language, String
+     */
+    public static void checkUcumCodings(List<Coding> ucumCodings, String elementName) {
+        if (ucumCodings != null) {
+            for (Coding ucumCoding : ucumCodings) {
+                checkUcumCoding(ucumCoding, elementName);
+            }
+        }
+    }
+    
+    /**
+     * Checks that the UCUM coding has a valid UCUM syntax.
+     * @param ucumCoding the UCUM coding
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM coding is not valid
+     */
+    public static void checkUcumCoding(Coding ucumCoding, String elementName) {
+        if (ucumCoding != null) {
+            if (hasSystemAndCodeValues(ucumCoding)) {
+                if (!UCUM_CODE_SYSTEM_URL.equals(ucumCoding.getSystem().getValue())) {
+                    throw new IllegalStateException(String.format("UCUM system is not '%s'", UCUM_CODE_SYSTEM_URL));
+                }
+                checkUcumCode(ucumCoding.getCode(), elementName);
+                return;
+            }
+            throw new IllegalStateException(String.format("'%s' does not contain a UCUM system of '%s' and a valid UCUM code", elementName, UCUM_CODE_SYSTEM_URL));
+        }
+    }
+
+    /**
+     * Checks that each UCUM codeable concept in the list has at least one coding with a valid UCUM syntax.
+     * @param ucumCodeableConcepts the UCUM codeable concept list
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM codeable concept list is not valid
+     */
+    public static void checkUcumCodeableConcepts(List<CodeableConcept> ucumCodeableConcepts, String elementName) {
+        if (ucumCodeableConcepts != null) {
+            for (CodeableConcept ucumCodeableConcept : ucumCodeableConcepts) {
+                checkUcumCodeableConcept(ucumCodeableConcept, elementName);
+            }
+        }
+    }
+
+    /**
+     * Checks that the UCUM codeable concept has at least one coding with a valid UCUM syntax.
+     * @param ucumCodeableConcept the UCUM codeable concept
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM codeable concept is not valid
+     */
+    public static void checkUcumCodeableConcept(CodeableConcept ucumCodeableConcept, String elementName) {
+        if (ucumCodeableConcept != null) {
+            if (ucumCodeableConcept.getCoding() != null) {
+                for (Coding coding : ucumCodeableConcept.getCoding()) {
+                    if (coding != null) {
+                        try {
+                            checkUcumCoding(coding, elementName);
+                            return;
+                        }
+                        catch (IllegalStateException e) {}
+                    }
+                }
+            }
+            throw new IllegalStateException(String.format("'%s' does not contain a Coding with a UCUM system of '%s' and a valid UCUM code", elementName, UCUM_CODE_SYSTEM_URL));
```

Comment:
when the for loop succeeds and the coding is checked it can throw an illegalState?  I don't think it should return  elementName as part of the String.format, it's difficult to  back track in the error to the actual failing element

Reply:
In this case we're just looking for one valid Coding. The checkUcumCoding method will return an IllegalStateException if a coding is not valid, so we want to just catch the exception and continue processing codings until we find a valid one, at which point we return from the method. If no valid codings are found we drop out of the loop and throw the IllegalStateException. For the error msg, do you suggest I just omit the CodeableConcept element name, or should I be providing something else there?